projects/ad7768-1fmcz/src/ad77681evb.c

```diff
@@ -133,31 +131,31 @@ int main()
 			printf("\r\n");
 			mdelay(1000);
 		}
-	} else { // offload example
-		msg = (spi_eng_msg *)malloc(sizeof(*msg));
-		if (!msg)
-			return -1;
+	} else {
+		ret = spi_engine_offload_init(spi_eng_desc, &spi_engine_offload_init_param);
+		if (ret != SUCCESS)
+			return FAILURE;
 
-		msg->spi_msg_cmds = malloc(sizeof(spi_msg_cmds));
-		msg->spi_msg_cmds = spi_msg_cmds;
-		msg->rx_buf_addr = 0x800000;
-		msg->tx_buf_addr = 0xA000000;
-		msg->msg_cmd_len = sizeof(spi_msg_cmds) / sizeof(uint32_t);
-		msg->tx_buf[0] = AD77681_REG_READ(AD77681_REG_ADC_DATA);
-		msg->tx_buf[1] = 0x00;
+		spi_engine_offload_message.commands = spi_msg_cmds;
+		spi_engine_offload_message.no_commands = ARRAY_SIZE(spi_msg_cmds);
+		spi_engine_offload_message.commands_data = commands_data;
+		spi_engine_offload_message.rx_addr = 0x800000;
+		spi_engine_offload_message.tx_addr = 0xA000000;
 
-		spi_eng_offload_load_msg(adc_dev->spi_desc, msg);
-		spi_eng_transfer_multiple_msgs(adc_dev->spi_desc, 8);
+		ret = spi_engine_offload_transfer(spi_eng_desc, spi_engine_offload_message,
+						  AD77681_EVB_SAMPLE_NO);
+		if (ret != SUCCESS)
+			return ret;
 
-		data = (uint8_t*)adc_dev->spi_desc->rx_dma_startaddr;
+		mdelay(1000);
 
-		mdelay(10000);
+		Xil_DCacheInvalidateRange(spi_engine_offload_message.rx_addr,
```

Comment:
Shouldn't #ifdef XILINX_PLATFORM be used for this function and 	Xil_ICacheEnable();	Xil_DCacheEnable();
like in other projects?

Reply:
`XILINX_PLATFORM ` comes from an `app_config.h` in other projects. ad7768-1fmcz project, it doesn't have such a config file yet. Probably it should be added in a future commit.